stellargraph/core/graph.py

```diff
@@ -302,6 +300,27 @@ def __init__(
             nodes=self._nodes,
         )
 
+    @staticmethod
+    def _infer_nodes_from_edges(edges, source_column, target_column):
+
```

Comment:
I think it'd be good to have some sort of comment about not emitting errors here like in each of my earlier suggestion, for a few reasons:

- the reason for doing so isn't obvious
- we want to retain this behaviour
- it's not easy to communicate this requirement directly in the code, so someone (like you or me) editing it in future may not realise or remember about this requirement

It's true that there's tests, but those are (conceptually) far way, and don't directly connect to this `_infer_nodes_from_edges` function. (For instance, those tests would pass if `_infer_nodes_from_edges` duplicated all of the error handling logic.)

Here's another option if you didn't like the earlier one:

```suggestion
        # `convert_edges` nicely flags any errors in edges; inference here is lax rather than duplicate that
```